### ISSUE TITLE
isPlainObject function is wrong

### ISSUE THREAD

**Issue by zozzz** [survivejs/webpack-merge#144]
Hi, this function is wrong: 
https://github.com/survivejs/webpack-merge/blob/2e8780ead820b9b2eecb13f394613acf325c0d91/src/join-arrays.ts#L63-L65
because returns `true` when `a` is `null` so currently, `isPlainObject(null) === true`. Why not use [is-plain-object](https://www.npmjs.com/package/is-plain-object) package instead?

**Comment by bebraw** [comment]
I went with the minimal amount of external dependencies.

Can you send me a test case where the current check is problematic? Simple configurations to merge would do.

**Comment by zozzz** [comment]
```javascript
const { merge } = require("webpack-merge")

const a = {
    devServer: { base: true }
}

const b = {
    devServer: null
}

console.log(merge(a, b))
```

expected result: `{ devServer: null }`

**Comment by bebraw** [comment]
Fixed in `5.1.2`. Thanks!
